Add aggregations/CCS memory leak as a 7.14.x known issue

ppf2 opened the pull request:
Adding this as a known issue (https://github.com/elastic/elasticsearch/pull/78404) to the release notes for 7.14.x (fix will be in 7.15.1). Thx!

ppf2 committed:
Add aggregations/CCS memory leak as a 7.14.x known issue

There is currently no plans to back port https://github.com/elastic/elasticsearch/pull/78404 to 7.14.x.
Adding this as a known issue to the release notes for 7.14.x per discussion with @qhoxie .  Please also back port this doc change to 7.14.0 and 7.14.1. Thx!

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

henningandersen commented on docs/reference/release-notes/7.14.asciidoc:
```diff
@@ -3,6 +3,22 @@
 
+[[known-issues-7.14.2]]
+[discrete]
+=== Known issues
```
I think we should include in the text that the issue exists since 7.14.0?

henningandersen replied:
I did not find this in the text, only in the anchor above. I think we should simply call it out in the text below.

henningandersen commented on docs/reference/release-notes/7.14.asciidoc:
```diff
@@ -3,6 +3,22 @@
+configured for cross cluster search (sniff and proxy). In sniff mode, we only connect to a subset of the 
+remote nodes (by default 3). So if the remote node we want to send a request to is not one of those 3, 
+we must send the request as a proxy request. The workaround is to periodically restart nodes with heap pressure 
+or cancel the aggregation tasks on the nodes.
```
This last bit is not true, there is no way to cancel this task, it's execution has completed but left a leak.

ppf2 replied:
good to know thx!

henningandersen commented on docs/reference/release-notes/7.14.asciidoc:
```diff
@@ -3,6 +3,22 @@
+* Aggregations: When a cross cluster search request is proxied, the memory for the aggregations on the
+proxy node will not be freed. The trigger is cross cluster search using aggregations where minimize 
+roundtrips is not effective (e.g. when minimize roundtrips is explicitly disabled, or implicitly disabled 
+when using scroll, async and point in time searches). This issue can happen in all modes of remote connections 
```
I think we should note that this affects most searches done from kibana?

henningandersen reviewed (commented):
Thanks @ppf2 I left a couple of comments, otherwise looking good.

[preview](https://elasticsearch_78728.docs-preview.app.elstc.co/diff)

ppf2 committed:
Update 7.14.asciidoc

ppf2 committed:
Update 7.14.asciidoc

Updated with 7.14.0-7.14.2 reference and explicitly call out its impact on Kibana CCS+aggregation requests.

ppf2 committed:
Update 7.14.asciidoc

ppf2 committed:
Update 7.14.asciidoc

ppf2 committed:
Create 7.14.asciidoc

Removed task cancellation as a workaround.

ppf2 commented:
@henningandersen Updated with your suggestions, thx!  I will make the same changes (Kibana callout and removing task cancellation workaorund) to the 7.15.0 release notes PR.
@tbrooks8 Would you like to review as well?

henningandersen commented on docs/reference/release-notes/7.14.asciidoc:
```diff
@@ -3,6 +3,22 @@
 
 Also see <<breaking-changes-7.14,Breaking changes in 7.14>>.
 
+[[known-issues-7.14.0 to 7.14.2]]
```
I think this does not work, see the [preview](https://elasticsearch_78728.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.14/release-notes-7.14.2.html).

ppf2 replied:
Nice, I have always wondered how I can preview doc changes on our side.  Thx for the link!  I will remove the anchor and just have the info inline :)

henningandersen reviewed (commented):
Thanks @ppf2 , a couple more comments.

ppf2 committed:
Update 7.14.asciidoc

Removed anchor and placing affected versions inline.

ppf2 committed:
Update 7.14.asciidoc

fixing header

henningandersen reviewed (approved):
LGTM

jrodewig committed:
Move version number to text. Add attributes.

jrodewig committed:
Minor tweaks

jrodewig committed:
Reuse in all 7.14.n RNs

jrodewig reviewed (approved):
Thanks for raising this @ppf2.

There are some wording improvements we can make here, but the current copy is clear enough; I don't want to hold this up any longer than needed.

I pushed some small commits to use attributes and reuse the copy in all 7.14 RNs. I'll get this merged and forward-ported to the 7.15 and 7.x branches.
